rosbag2_py/test/test_sequential_writer.py

```diff
@@ -48,13 +47,9 @@ def test_sequential_writer():
 
     :return:
     """
-    bag_path = str(Path(__file__).parent.parent / 'resources' / 'tmp_write_test')
-
-    if os.path.exists(bag_path):
-        print('%s exists. It will be overwritten by new rosbag2_py test_sequential_writer run.' % (
-            bag_path))
-        shutil.rmtree(bag_path)
-    os.mkdir(bag_path)
+    with tempfile.TemporaryDirectory() as tmpdir:
+        bag_path = os.path.join(tmpdir, 'tmp_write_test')
+    os.makedirs(bag_path)
```

Comment:
nitpick: the TemporaryDirectory context will create the path for us. We can indent the remainder of the test to ensure the directory exists for the entire test.

I didn't think of it before, but alternatively we could use the [pytest fixture `tmp_path`](https://docs.pytest.org/en/stable/tmpdir.html) and avoid indentation altogether.